core/src/main/java/com/graphhopper/routing/weighting/custom/ExpressionVisitor.java

```diff
@@ -40,7 +37,13 @@
     private final Set<String> allowedMethods = new HashSet<>(Arrays.asList("ordinal", "getDistance", "getName",
             "contains", "sqrt", "abs"));
     private String invalidMessage;
-    private DefaultEncodedValueFactory factory = new DefaultEncodedValueFactory();
+    // todo: not sure if this is worth it just to slightly improve the error message
+    private final Set<String> knownEncodedValues = new HashSet<>(Arrays.asList(
+            Roundabout.KEY, "car_access", "bike_access", GetOffBike.KEY, RoadClass.KEY, RoadClassLink.KEY, RoadEnvironment.KEY,
+            RoadAccess.KEY, MaxSpeed.KEY, MaxWeight.KEY, MaxWidth.KEY, MaxAxleLoad.KEY, MaxLength.KEY, Surface.KEY, Smoothness.KEY,
+            Toll.KEY, TrackType.KEY, BikeNetwork.KEY, FootNetwork.KEY, Hazmat.KEY, HazmatTunnel.KEY, HazmatWater.KEY,
+            Lanes.KEY, MtbRating.KEY, HikeRating.KEY, HorseRating.KEY, Country.KEY
+    ));
```

Comment:
@karussell: If I got this right we used the `DefaultEncodedValueFactory` here to distinguish between 'invalid identifiers' and names of encoded values that are known to the factory, but were not added to the graph. The only purpose of this seem to be improved error messages. I replaced the hard-coded factory with a hard-coded list of encoded value names here. But actually I think we should consider removing this list. A given graphhopper instance has a list of encoded values stored in the graph. I don't think we should maintain a list of encoded values that aren't there, but could be there... At least not until we use such a list of encoded values that could be there (a registry in a way) to actually create the encoded values. Anyway, I updated this list according to: https://github.com/graphhopper/graphhopper/issues/2438#issuecomment-948746635.

Reply:
Yes, sure. Let's remove this.

Reply:
Yes, sure. Let's remove this.